test/Microsoft.AspNet.Mvc.Core.Test/ControllerTests.cs

```diff
@@ -424,6 +424,34 @@ public void Created_WithRelativeUriParameter_SetsCreatedLocation()
         }
 
         [Fact]
+        public void Created_IDisposableObject_RegistersForDispose()
+        {
+            // Arrange
+            var mockHttpContext = new Mock<DefaultHttpContext>();
+            mockHttpContext.Setup(x => x.Response.OnResponseCompleted(It.IsAny<Action<object>>(), It.IsAny<object>()))
+                .Verifiable();
+            var uri = new Uri("/test/url", UriKind.Relative);
+
+            var controller = new TestableController()
+            {
+                ActionContext = new ActionContext(mockHttpContext.Object, new RouteData(), new ActionDescriptor())
+            };
+            var input = new DisposableObject();
```

Comment:
This test just checks if this was registered. It is not called. We can write a functional test and verify if it gets called by TestHost.